types/setting.go

```diff
@@ -1371,44 +1454,7 @@ func ValidateSetting(name, value string) (err error) {
 			return fmt.Errorf("value %v of setting %v should be true or false", value, sName)
 		}
 
-	case SettingNameStorageOverProvisioningPercentage:
```

Comment:
Suggest having a general `validateInt` and run it for each setting at the beginning. If the updated setting is not int type, then bypass. 

In `validateInt`, it will check if the setting in the range no matter if it just has min or max, and provide a consistent validation error message. This will simplify the code.

P.S. If we have more validation later, then it can be a validation chain like middle layers for validation.

Reply:
Fixed, thanks